src/commands.cpp

```diff
@@ -1457,67 +1464,75 @@ bool CommandMoveNode::procresult(Result r)
 
                 if(sync)
                 {
+                    // todo: is this already done by the actionpackets.  And should we have checked the error code
                     client->app->syncupdate_remote_move(sync, n, client->nodebyhandle(pp));
                 }
             }
         }
 #endif
         // Movement of shares and pending shares into Rubbish should remove them
-        if (r.wasError(API_OK))
+        if (!e)
         {
-            Node *n = client->nodebyhandle(h);
-            if (n && (n->pendingshares || n->outshares))
-            {
-                Node *rootnode = client->nodebyhandle(np);
-                while (rootnode)
-                {
-                    if (!rootnode->parent)
-                    {
-                        break;
-                    }
-                    rootnode = rootnode->parent;
-                }
-                if (rootnode && rootnode->type == RUBBISHNODE)
-                {
-                    share_map::iterator it;
-                    if (n->pendingshares)
-                    {
-                        for (it = n->pendingshares->begin(); it != n->pendingshares->end(); it++)
-                        {
-                            client->newshares.push_back(new NewShare(
-                                                            n->nodehandle, 1, n->owner, ACCESS_UNKNOWN,
-                                                            0, NULL, NULL, it->first, false));
-                        }
-                    }
-
-                    if (n->outshares)
-                    {
-                        for (it = n->outshares->begin(); it != n->outshares->end(); it++)
-                        {
-                            client->newshares.push_back(new NewShare(
-                                                            n->nodehandle, 1, it->first, ACCESS_UNKNOWN,
-                                                            0, NULL, NULL, UNDEF, false));
-                        }
-                    }
-
-                    client->mergenewshares(1);
-                }
-            }
+            // todo: is this really done by the actinpacket code
```

Comment:
Trying the same thing (moving the outgoing share to the rubbish) in the webclient, logged in with the same user, does update megacli's share info.   It's removed by an s2 actionpacket.   It turns out the webclient is sending` s2` before it sends `m` in that case.   I think we should modify the SDK to do the same, and probably we should do that as a separate PR to develop?     
For this PR, I'll remove this commented block.

Reply:
Not sure if it should be a separate PR. The change in the logic comes from this PR, which removes the code to _unshare_ the folder that is moved to the rubbish bin. Hence, rather than removing that code and introducing a bug, should we keep consistency and adjust the behavior (`s2`+`m` instead of than `m`+`s2`) as part of this branch?
Actually, the ideal would be that, if MEGA considers that any shared folder that is moved into the rubbish bin should stop sharing, it should be the API who automatically _unshares_ the folder (similar to the automatic _unshare_ when a node is deleted, or the target contact is removed, or the sharer's account is cancelled...)

What do you think?
1. Change `m`+`s2` --> `s2`+`m` as part of this PR.
2. Change `m`+`s2` --> `s2`+`m` as part of a new ticket / another PR to develop.
3. Ask API to automatically unshare shared folders moved into the rubbish bin.

Reply:
I've checked with API team, they would prefer us to go with 1 for now, but keep 3 in mind for the future.   I'll prepare a PR to this branch.